dialect/sql/builder.go

```diff
@@ -254,7 +254,11 @@ func (t *TableAlter) DropColumn(c *ColumnBuilder) *TableAlter {
 
 // AddForeignKey adds a foreign key constraint to the `ALTER TABLE` statement.
 func (t *TableAlter) AddForeignKey(fk *ForeignKeyBuilder) *TableAlter {
-	t.Queries = append(t.Queries, &Wrapper{"ADD CONSTRAINT %s", fk})
+	if t.Dialect() == dialect.SQLite {
+		t.Queries = append(t.Queries, &Wrapper{"ADD COLUMN %s", fk})
+	} else {
+		t.Queries = append(t.Queries, &Wrapper{"ADD CONSTRAINT %s", fk})
+	}
```

Comment:
IMO this logic should reside in the schema migration package.

I'm thinking about 2 options (but you're welcome to suggest other alternatives):
1. Add an `addFK` method to the `sqlDialect` interface that generates a different command for the `sqlite3` dialect.
2. Change `fkExist` logic in `sqlite3` to always returns `true` (as it was before), and append the [foreign-key-clause](https://sqlite.org/syntax/foreign-key-clause.html) to the new column we create. For example:
   ```sql
   ALTER table t1 ADD COLUMN c1 int REFERENCES t2(c2) ON DELETE SET NULL;
   ```

Let me know what do you think about it.

Reply:
ok, that makes sense. I think I get the general gist of what your'e suggesting:

```go
func (s *SQLite) addFK(t *Table, *fk ForeignKey) string {
   // construct string to add foreign key
}

func (s *SQLite) fkExist(t *Table, *fk ForeignKey) bool {
   return true
}
```

Reply:
@a8m ok after some noodling on this (and re-reading what you said, multiple times, sorry about missing the message there!):

* addFK leads to import cycles best I can see without creating a third package, but I'm still playing around with this approach and it might prove fruitful. I can't share the migrate package with the builder directly because of the relationship that already exists. I'm guessing I'm missing something here and will report back after more searching around.

* setting things to true works fine, but then the alter cannot work at all unless I am still misunderstanding? I'm not sure is the best approach to adding edges to existing tables. I've implemented most of this and the clear problem with it is that it will never work on the second migration to the same table as the user probably would expect (with fks populated).

I think what might make sense here (but need to examine a little more) is altering the addColumn interface method to accept a foreignkey struct, or otherwise relate the foreign keys to the columns being currently built (which is true for FKs but not the columns that they are related to). This way I can deliberately say "I want to ensure my foreign key relationships are always added as a part of my column mutations to tables, no other way". And I think, but am not 100% certain, this way the work in fk.DSL() can either be simplified or removed for SQLite's case and we can just match/query the foreign keys directly in the Column portion of the migration in the tBuilder method and simplify the building process here.

WDYT?

Also sincerely sorry for the uber-high latency on this patch, I'll try to button things up this week.

Reply:
> I think what might make sense here (but need to examine a little more) is altering the addColumn interface method to accept a foreignkey struct, or otherwise relate the foreign keys to the columns being currently built (which is true for FKs but not the columns that they are related to).

I think that's fine (as I mentioned in https://github.com/facebookincubator/ent/pull/221#discussion_r361163803). If you want to go with this approach, I can help with that and add the code that links the columns with their foreign-keys.  

WDYT?